### DISCUSSION TITLE
Align versions also on Dockerhub

### DISCUSSION THREAD

**Discussion by kaulex99** [nextcloud/all-in-one#2851] [Ideas] [answered]
Currently, there are no exact version tags on Dockerhub which may lead to confusion in the community. Would it be possible to achieve that in the near future?

**Comment by szaimen** [comment]
Hi, what kind of tags would you like to see on docker hub? Do you have an example?

**Reply by Zoey2936** [reply]
I think they mean something like: nextcloud/all-in-one:6.1.1

**Reply by kaulex99** [reply]
My first thought was the current Nextcloud version associated. But then I also thought about the case that it might/might not be that there is an AIO update without a Nextcloud update. IMO, preferable Nextcloud Version, but if not possible because of independent AIO updates, the AIO version - or a combination of both? 

Just suggestions. I just wanted to start a discussion about that here :)

**Reply by szaimen** [reply]
I see and why would you need this?

**Reply by kaulex99** [reply]
The idea behind was, that I want to have control about the version which I want to install. Furthermore, I would like to i.e. install v6.2.1 (which is currently a pre-release and not in latest AFIAK) without changing to beta and then back again to latest. And I think this is the default behaviour in nearly all dockerhub projects - so why not here. But maybe I miss a contra.

**Comment by szaimen** [comment, marked as answer]
> But maybe I miss a contra.

Yes. The main reason ist that in that case all update functionalty of AIO will not work anymore like mastercontainer and container updates and updatenotification. So we don't actually want people to pin the version as they will need to always change the tag and update (recreate) the mastercontainer and then also all other containers afterwards manually which is likely not feasible for many beginners. And I fear many beginners will use such a tag when we have them because they dont know better.

However you can use the tags like e.g. 20230613_120442-latest to pin the AIO version if you really need to which is definitely an advancesd use case and should  not be the standard for AIO users.

**Reply by kaulex99** [reply]
But than it is already the case also for new users. Providing a 6.1.1 tag would be just a more readable tag. And if not announced widely in the installation guides why would new users go with that if they don't know about it?

**Reply by szaimen** [reply]
> But than it is already the case also for new users. Providing a 6.1.1 tag would be just a more readable tag.

Yeah, it might be more readable but nothing more.

> And if not announced widely in the installation guides why would new users go with that if they don't know about it?

The problem is that people simply do not read the documentation.

**Comment by gggeek** [comment]
> The main reason ist that in that case all update functionalty of AIO will not work anymore like mastercontainer and container updates and updatenotification

I am not sure I get that - maybe I miss the deep technical reason details.

As long as users start the MC from `latest`, they get everything working as expected.

People who want to run specific version of either the MC or the app containers should be free to do so with manual-install and such?
